src/main/java/uk/gov/hmcts/reform/blobrouter/services/report/ReportService.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.hmcts.reform.blobrouter.services.report;
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.blobrouter.data.ReportRepository;
+import uk.gov.hmcts.reform.blobrouter.model.out.DailyReportResponse;
+import uk.gov.hmcts.reform.blobrouter.model.out.EnvelopeSummaryResponse;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static java.time.LocalDateTime.ofInstant;
+import static java.time.ZoneOffset.UTC;
+import static java.util.stream.Collectors.toList;
+import static uk.gov.hmcts.reform.blobrouter.data.model.Status.DISPATCHED;
+import static uk.gov.hmcts.reform.blobrouter.data.model.Status.REJECTED;
+
+@Service
+public class ReportService {
+    public static final String TIME_FORMAT = "HH:mm:ss";
+    private final ReportRepository reportRepository;
+
+    public ReportService(ReportRepository reportRepository) {
+        this.reportRepository = reportRepository;
+    }
+
+    public DailyReportResponse getDailyReport(LocalDate date) {
+
+        Instant from = date.atStartOfDay().toInstant(UTC);
+        Instant to = date.atStartOfDay().plusDays(1).toInstant(UTC);
+
+        AtomicInteger rejected = new AtomicInteger(0);
+        AtomicInteger dispatched = new AtomicInteger(0);
+        List<EnvelopeSummaryResponse> envelopeSummaryResponses =
+            reportRepository.getEnvelopeSummary(from, to).stream()
+                .map(s -> {
+                    if (s.status == DISPATCHED) {
+                        dispatched.getAndIncrement();
+                    } else if (s.status == REJECTED) {
+                        rejected.getAndIncrement();
+                    }
+                    return new EnvelopeSummaryResponse(
+                        s.container,
+                        s.fileName,
+                        ofInstant(s.fileCreatedAt, UTC).toLocalDate(),
```

Comment:
static import hides information here. 'of instant' what?

Reply:
Will fix